src/Dotnet.Script.Core/Interactive/InteractiveRunner.cs

```diff
@@ -70,8 +72,14 @@ public InteractiveRunner(ScriptCompiler scriptCompiler, ScriptLogger logger, Scr
                 }
                 else
                 {
-                    var lineDependencies = ScriptCompiler.RuntimeDependencyResolver.GetDependenciesFromCode(CurrentDirectory, input);
-
+                    var lineRuntimeDependencies =
+                        ScriptCompiler.RuntimeDependencyResolver.GetDependenciesFromCode(CurrentDirectory, input).ToArray();
+                    var lineDependencies = lineRuntimeDependencies.SelectMany(rtd => rtd.Assemblies).Distinct();
+
+                    var scriptMap = lineRuntimeDependencies.ToDictionary(rdt => rdt.Name, rdt => rdt.Scripts);
```

Comment:
did this work? btw, we should only reset the resolver when the `scriptMap` is not empty

Reply:
It did work, although I think the `InteractiveRunner` needs some more work.  I noticed that it takes another approach the first time it is executed, `RunFirstScript` . I did not want to mess with that too much since I guess there is a good reason for that special handling 😄

Reply:
btw, I also added a test, `ScriptPackageReference` in the `InteractiveRunnerTests` class for this

Reply:
yeah you are right, there is a good reason for this, but this is also a bug 😀 We can fix this in a separate PR

Reply:
Awesome 👍